mne/decoding/xdawn.py

```diff
@@ -160,23 +157,22 @@ def inverse_transform(self, X):
 
         Returns
         -------
-        out : dict of instance
-            A dict of instance (from the same type as inst input) for each
-            event type in event_id.
+        out : np.ndarray, shape(n_channels, n_times, n_freqs)
+            Array for each event type in event_id concatenated sequentially.
         """
 
         if not isinstance(X, np.ndarray):
             raise ValueError("Given data should be numpy array, got "
                              "%s instead" % type(X))
 
         data = np.hstack(X)
-        data_dict = dict()
+        result = []
         event_id = self.event_id
 
         for eid in event_id:
 
             data_r = self._pick_sources(data, None, None, eid)
             data_r = np.array(np.split(data_r, self._n_chan, 1))
-            data_dict[eid] = data_r
+            result.append(data_r)
 
-        return data_dict
+        return np.concatenate(result)
```

Comment:
don't you need to specify the axis? what's the dimensionality of this?

Is it tested in unit tests?